### PR TITLE
OP_PICK opcode implementation

### PR DESCRIPTION
**Pull request by KhairallahA**
OP_PICK implementation for #42

### CODE REVIEW

File: src/script/engine.zig [hunk 1]
```diff
@@ -414,6 +415,19 @@ pub const Engine = struct {
...
 
+    /// OP_PICK: The item idx back in the stack is copied to the top.
+    /// 
+    /// # Returns
+    /// -  "EngineError.StackUnderflow": if idx < 0
+    fn opPick(self: *Engine) !void {
+        const idx = try self.stack.popInt();
+        if (idx < 0) {
+            return error.StackUnderflow;
+        }
```
Comment (tdelabro) [inline src/script/engine.zig:426]: unecessary check `self.stack.peek` is already doing this exact same check

File: src/script/engine.zig [hunk 2]
```diff
@@ -595,6 +609,29 @@ test "Script execution - OP_1 OP_2 OP_DROP" {
     try std.testing.expectEqualSlices(u8, &[_]u8{1}, element0);
 }
 
+test "Script execution - OP_PICK" {
+    const allocator = std.testing.allocator;
+
+    // Simple script: OP_1 OP_1 OP_EQUAL
```
Comment (tdelabro) [inline src/script/engine.zig:615]: I think the comment don't match the script

File: src/script/stack.zig [hunk 3]
```diff
@@ -87,7 +87,7 @@ pub const Stack = struct {
 
         if (value.len > 8) return StackError.InvalidValue;
 
-        var buffer: [8]u8 = undefined;
+        var buffer: [8]u8 = [_]u8{0, 0, 0, 0, 0, 0, 0, 0};
         std.mem.copyBackwards(u8, buffer[0..value.len], value);
```
Comment (tdelabro) [inline src/script/stack.zig:91]: you can use memcopy there

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: src/script/stack.zig (hunk 3 above)
Comment (KhairallahA) [reply src/script/stack.zig]: @tdelabro can you check now from the updates. Is that what you mean?

**Comment by tdelabro** [comment]
@KhairallahA don't forget to ask for new review on gh when you are done with your changes

**Review by tdelabro — APPROVED** [review]

**Pull request merged**